### ISSUE TITLE
No-code experiments / Visual editor

### ISSUE THREAD

**Issue by annikaschmid** [PostHog/posthog#16417]
The Feature Success team is currently considering adding support for no-code experiments. This would allow users to test changes to their website / app through a visual editor, without having to touch the code.

🗂 Considered Goals

- [ ] Allow making changes to text
- [ ] Allow making changes to styling
- [ ] ...

❓Open questions

- Is this a use case limited to experiments, or are users also interested in having a visual editor for feature flags?

We'd love to hear user feedback, as well as ideas for stretch goals or implementation. Please add to this issue!

If you like this idea, please leave a 👍 or ❤️ reaction on this post to vote for it -- your votes and feedback help us prioritize what to work on next!

**Comment by jackpordi** [comment]
# Why

My team, like many others, are currently working on finding a replacement to Google Optimize due to its sunset date.

We’ve currently narrowed down our replacement options to VWO and PostHog. Whilst initially the scope of this “replacement” was just to replace the A/B experiment platform, it does look like PostHog can replace a whole lot more services (that we’re either using now, or will do in the future) for a similar/competitive price. However, one major disadvantage of PostHog is also that our non-technical staff often do A/B experiments themselves on marketing copy/imagery (and sometimes even HTML/CSS edit themselves) without any help from the engineering team.

On the surface level, we could just “take the hit” temporarily if PostHog were to commit to building a visual editor within a certain timeline. However, we build our marketing/static pages using a headless CMS, so in a lot of places we can’t directly just “go into the code” and use the feature flags to change content directly there either, we’d need to build something which integrates into our CMS directly (and we also use the CMS in a number of different ways, depending on the type of page, so this could be a lot of effort). I can certainly explore further into any other workarounds we may have for this, but with a visual editor this would have been a slam-dunk for PostHog (especially given the number of people who are having to migrate off Optimize due to the sunset).

So far, everyone on our team has been extremely impressed by PostHog, and are very keen on using it (particularly, not just as an experimentation platform but as a full replacement for our analytics suite). The lack of visual editor is probably the only real major blocker here, as we’re a very growth-driven startup that needs to be able to do experiments quickly (relying on the fact that non-engineers can perform them).

# What would we like to see in a visual editor?

- The ability to click on some text on a page, and change it.
- The ability to click on an image on a page, and change it.
- The ability to click on any element on a page, and edit its HTML/CSS directly. This may also include creating new HTML elements and CSS groups from scratch.

**Comment by meikitanious** [comment]
I know personally we'd love to have a basic editor similar to Google Optimize; would be great for feature flags and A/Bs.

**Comment by utkuzih** [comment]
To add to this, allowing page redirects for certain variants on certain pages would help us test large changes. For example: 

### What to test
New hero-section layout on https://posthog.com/ (not a copy change but a complete visual re-haul).

### How to test?
If a user is on `/` page and is in `test` group, redirect user to `/new-hero-layout`

### Workaround
We can currently do the above by writing the following custom code on the tested page.

```
<script>
var featureFlag = 'landing_page_new_hero_layout';
var redirectPage = '/new-hero-layout';
var variantName = 'test';

window.addEventListener('posthog:loaded', function() {
  posthog.onFeatureFlags(function () {
    if (posthog.getFeatureFlag(featureFlag) === variantName) {
      window.location.href = redirectPage
    }
  });
}, false);
</script>
```

**Comment by lunchpaillola** [comment]
I'd definitely use a no-code interface of this sort! On a related no-code note, I think deeper support for Webflow and Bubble through a native integration (bubble plugins, webflow apps) would be really valuable. 

Deep analytics are a challenge on these platforms, and while there are partial third-party solutions, they're not always reliable. A PostHog-approved solution could really simplify things

**Comment by LukasHechenbergerID** [comment]
Hello!
I wanted to know if there is some development in considering this?

We switched our tech stack to Storyblok as CMS and Nuxt for the Frontend (Posthog was implemented earlier and is running on all kinds of different sites with different technologies). 
Because of the freedom this provides for our people that build the pages it is quite hard to A/B Test.
The only feasible way we can think of is to switch whole sections.

But this would be ideal to test small text changes or small style changes (adding inline styles to elements?).
For the start it would be great if we can do this and of course easily connect these changes to experiments.

It would also help greatly to do A/B Tests on our different sites with different technologies.

**Comment by jgbishop6262** [comment]
Similar to Lukas above, I know my team would really value being able to run split tests via a CMS (our design team is heavily using Builder.io lately). Being able to set up different pages/modules that render based on a test and flag this without code would unblock a lot of rapid iteration for the UX team.

**Comment by stuartbrameld** [comment]
We're currently paying for convert.com and would love to have this all in one place with PostHog. For marketing teams many of these ab testing tools are actually super clunky, particularly when it comes to managing conversion events, hence being able to have this all in one place would be a big advantage.

**Comment by annikaschmid** [comment]
Hi everyone! Thanks for everyone's feedback so far. Good news: We are going to start working on no-code experiments this month. @Phanatic is leading here, and I am supporting on the Product side. 🚀

If there is any additional feedback you would like to share with us, or specify your requirements, you can leave a comment on this post, or book a short call with us here: https://cal.com/posthog-annika/no-code-experiments

We will keep you updated in this issue when a Beta is available. Thanks!

**Comment by hotairbag** [comment]
would love to try this!

**Comment by stuartbrameld** [comment]
likewise would love to beta test 🚀

**Comment by annikaschmid** [comment]
Hi everyone, we have a first version of no-code experiments available in private Beta. If you want to try it out, please email me at annika AT youcanguesstherest. We will add you to the feature flag, and to a Slack channel where you can share feedback and missing features. Thanks!

**Comment by torchsmith** [comment]
I just want to mention this issue here to link it: https://github.com/PostHog/posthog/issues/30183
